In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pymysql
import getpass 
import statsmodels.api as sm

In [2]:
sns.set_style("white")

In [3]:
price = np.array([7,20,13,59,40,51,37])
review_score = np.array([7,6,8,1,3,2,3])

### **1. Mean**

In [5]:
type(price)

numpy.ndarray

In [6]:
price.mean()

32.42857142857143

In [7]:
review_score.mean()

4.285714285714286

### **2. Variance**

In [8]:
ddof = 1

In [11]:
price.var(ddof=ddof)

384.61904761904754

In [10]:
review_score.var(ddof=ddof)

7.238095238095238

### **3. Standard Deviation**

In [12]:
price.std(ddof=ddof)

19.611706902231827

In [13]:
np.sqrt(price.var(ddof=ddof))

19.611706902231827

In [14]:
review_score.std(ddof=ddof)

2.690370836538197

In [15]:
np.sqrt(review_score.var(ddof=ddof))

2.690370836538197

### **4. Covariance**

In [16]:
np.cov(price, review_score, ddof=ddof)

array([[384.61904762, -51.30952381],
       [-51.30952381,   7.23809524]])

In [17]:
cov = np.cov(price, review_score, ddof=ddof)

In [18]:
cov[0,1]

-51.3095238095238

### **5. Correlation (Pearson)**

In [19]:
cov[0,1] / np.sqrt(price.var(ddof=ddof) * review_score.var(ddof=ddof))

-0.9724570819188173

In [20]:
np.corrcoef(price, review_score)

array([[ 1.        , -0.97245708],
       [-0.97245708,  1.        ]])

In [21]:
#use statsmodels
df = pd.DataFrame({"y": review_score, "x": price})

In [22]:
df

y   x
0  7   7
1  6  20
2  8  13
3  1  59
4  3  40
5  2  51
6  3  37

In [23]:
df = sm.add_constant(df)

In [24]:
df

const  y   x
0    1.0  7   7
1    1.0  6  20
2    1.0  8  13
3    1.0  1  59
4    1.0  3  40
5    1.0  2  51
6    1.0  3  37

In [25]:
#way 1
model = sm.OLS(df["y"], 
       df[["const", "x"]])
results = model.fit()

In [26]:
# way 2
results2 = sm.OLS(df["y"], 
                  df[["const", "x"]]).fit()

In [ ]:
#way 3
y = df["y"]
X = df[["const", "x"]]
lin_reg = sm.OLS(y, X).fit()

In [27]:
type(results)

statsmodels.regression.linear_model.RegressionResultsWrapper

In [28]:
results.summary()

/Users/emilyhorton/opt/anaconda3/envs/matrix/lib/python3.8/site-packages/statsmodels/stats/stattools.py:70: ValueWarning: omni_normtest is not valid with less than 8 observations; 7 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.946
Model:                            OLS   Adj. R-squared:                  0.935
Method:                 Least Squares   F-statistic:                     87.03
Date:                Mon, 07 Sep 2020   Prob (F-statistic):           0.000238
Time:                        10:57:18   Log-Likelihood:                -6.1262
No. Observations:                   7   AIC:                             16.25
Df Residuals:                       5   BIC:                             16.14
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          8.6118      0.531     16.204      0.000       7.246       9.978
x             -0.1334      0.014     -9.329      0.000      -0.170      -0.097
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   1.559
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.457
Skew:                           0.582   Prob(JB):                        0.796
Kurtosis:                       2.539   Cond. No.                         76.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

For increase in price by 1 EUR, our review_score decreases by 0.1334